File: Leetcode/1874/1874.cpp
class Solution {
public:
    int minProductSum(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end(),greater<>());
        int sum = 0;
        int n = (int)nums1.size();
        for (int i = 0; i < n; ++i) {
            sum += nums1[i]*nums2[i];
        }
        return sum;
    }
};

File: Leetcode/1874/1874.go
func minProductSum(nums1 []int, nums2 []int) int {
	sort.Ints(nums1)
	sort.Ints(nums2)
	sum := 0
	for i := 0; i < len(nums2); i++ {
		sum += nums1[i] * nums2[len(nums2)-i-1]
	}
	return sum
}